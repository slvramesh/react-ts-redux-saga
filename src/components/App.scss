
.App {
  .content{
    padding: 50px 0;
  }

  .MuiLinearProgress-colorPrimary{
    position: absolute;
    width: 100%;
    left: 0;
    right: 0;
  }
}