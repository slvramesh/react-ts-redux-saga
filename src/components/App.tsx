import React from 'react';
import { Route, BrowserRouter as Router } from 'react-router-dom';

import Container from '@material-ui/core/Container';

import ManageContact from './contact/ManageContact';
import Navigation from './Navigation';

import './App.scss';

const Home: React.FC = () => {
  return (
    <h2>Home page</h2>
  );
}

const users: React.FC = () => {
  return (
    <h2>Users page</h2>
  );
}


const App: React.FC = () => {
  return (
    <Container maxWidth="lg">
      <Navigation />
      <Router>
        <div className="App">
          <Route default exact path="/" component={Home} />
          <Route path="/contact" component={ManageContact} />
          {/* <Route path="/contact/:id" component={ManageContact} /> */}
          <Route path="/users" component={users} />
        </div>
      </Router>
    </Container>
  );
}

export default App;
