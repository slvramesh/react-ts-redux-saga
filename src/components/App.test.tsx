import React from "react";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import pretty from "pretty";

import App from "./App";

const mockStore = configureStore([]);

describe("App Component ->", () => {
  let store: any;
  let component: any;

  let container: any = null;
  beforeEach(() => {
    store = mockStore({});

    component = renderer.create(
      <Provider store={store}>
        <App />
      </Provider>
    );

    container = document.createElement("div");
    document.body.appendChild(container);
  });

  it("should render", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("should render a App", () => {
    act(() => {
      render(
        <Provider store={store}>
          <App />
        </Provider>,
        container
      );
    });

    expect(pretty(container.innerHTML)).toMatchInlineSnapshot(`
      "<div class=\\"MuiContainer-root MuiContainer-maxWidthLg\\">
        <div class=\\"App\\">
          <div class=\\"content\\">
            <div class=\\"contactForm\\">
              <div class=\\"addCon\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\">Add</span><span class=\\"MuiTouchRipple-root\\"></span></button></div>
            </div>
            <div class=\\"contactList\\">
              <p>No records!</p>
            </div>
            <div></div>
          </div>
        </div>
      </div>"
    `);
  });
});

//import React from 'react';
// import { render } from '@testing-library/react';
// import { unmountComponentAtNode } from "react-dom";
// import { act } from "react-dom/test-utils";
//import ShallowRenderer from 'react-test-renderer'

// import App from './App';
// import ManageContact from './contact/ManageContact';
// import Container from '@material-ui/core/Container';

// // test('renders learn react link', () => {
// //   const { getByText } = render(<App />);
// //   const linkElement = getByText(/learn react/i);
// //   expect(linkElement).toBeInTheDocument();
// // });

// let container: any = null;
// beforeEach(() => {
//   // setup a DOM element as a render target
//   container = document.createElement("div");
//   document.body.appendChild(container);
// });

// afterEach(() => {
//   // cleanup on exiting
//   unmountComponentAtNode(container);
//   container.remove();
//   container = null;
// });

// it('should render correctly with title and description', () => {
//   const wrapper = ShallowRenderer(
//     <App />
//   );

//   expect(toJson(wrapper)).toMatchSnapshot();
// });

// it("renders with or without a name", () => {
//   act(() => {
//     render(<App />, container);
//   });
//   expect(container.textContent).toBe("Hey, stranger");

// });

// import * as ShallowRenderer from 'react-test-renderer/shallow';

// // in your test:
// const renderer = ShallowRenderer.createRenderer();
// renderer.render(<App />);
// const result = renderer.getRenderOutput();

//expect(result.type).toBe('div');
// expect(result.props.children).toEqual([
//   <Container maxWidth="lg">
//     <div className="App">
//       <ManageContact />
//     </div>
//   </Container>
// ]);

// it("renders with or without a name", () => {
//   act(() => {
//     render(<App />, container);
//   });
//   expect(container.textContent).toBe("Hey, stranger");

// });

// import ReactDOM from "react-dom";

// import "./App.scss";

// it("renders without crashing", () => {
//   const div = document.createElement("div");
//   ReactDOM.render(<App />, div);
//   ReactDOM.unmountComponentAtNode(div);
// });
