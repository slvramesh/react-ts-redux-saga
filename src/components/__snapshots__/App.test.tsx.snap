// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component -> should render 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthLg"
>
  <div
    className="App"
  >
    <div
      className="content"
    >
      <div
        className="contactForm"
      >
        <div
          className="addCon"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              Add
            </span>
          </button>
        </div>
      </div>
      <div
        className="contactList"
      >
        <p>
          No records!
        </p>
      </div>
      <div />
    </div>
  </div>
</div>
`;
